addons/account_payment/account_move_line.py

```diff
@@ -27,9 +27,12 @@ class account_move_line(osv.osv):
     _inherit = "account.move.line"
 
     # delegate to parent, used for local fields.function redefinition
-    def _amount_residual(self, cr, uid, ids, field_names, args, context=None):
-        return super(account_move_line, self)._amount_residual(
-            cr, uid, ids, field_names, args, context=context)
+    def _amount_to_pay(self, cr, uid, ids, field_names, args, context=None):
+        return {
+            id: value['amount_residual']
+            for id, value in self._amount_residual(cr, uid, ids, field_names, args,
+                                                   context=context).values()
```

Comment:
you should use `.items()` instead of `.values()`

Your test should also read the field (assert the value == 0)